install/kubernetes/cilium/templates/cilium-agent/validatingadmissionpolicy.yaml

```diff
@@ -0,0 +1,26 @@
+{{- if .Capabilities.APIVersions.Has "admissionregistration.k8s.io/v1alpha1/ValidatingAdmissionPolicy" }}
+---
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "http-tcp"
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+      - apiGroups:   ["cilium.io"]
+        apiVersions: ["v2"]
+        operations:  ["CREATE", "UPDATE"]
+        resources:   ["ciliumnetworkpolicies"]
+  validations:
+    - expression:
+        #ingress validating
+        (has(object.spec.ingress) && object.spec.ingress.exists(x, has(x.toPorts) && x.toPorts.exists(x, (has(x.ports) &&
```

Comment:
Could we organize this with newlines/indentation to make it easier to read? Nested parens can make this tricky.

Here's an example. This is a just an example, it might make more sense to combine some lines.
```
(
  has(object.spec.ingress) 
  && object.spec.ingress.exists(x, has(x.toPorts) 
    && x.toPorts.exists(x, 
      (
        has(x.ports) && (
          x.ports.exists(x, x.protocol == 'TCP') 
          && (
            has(x.rules) && (
              has(x.rules.http) || has(x.rules.kafka)
            )
          ) 
        ) && (
          has(x.rules) && !has(x.rules.dns)
        )
      )
    )
  )
)
```